docs/book/src/tasks/experimental-features/cluster-classes.md

```diff
@@ -0,0 +1,341 @@
+# ClusterClass
+
+The ClusterClass feature introduces a new way to create clusters which reduces boilerplate and enables flexible and powerful customization of clusters.
+ClusterClass is a powerful abstraction implemented on top of existing interfaces and offering a set of tools and operations to streamline cluster lifecycle management while maintaining the same underlying API.
+
+This tutorial covers using Cluster API and ClusterClass to create a Kubernetes cluster and how to perform a one-touch Kubernetes version upgrade.
+
+<aside class="note warning">
+
+<h1>Note</h1>
+
+This guide covers creating a ClusterClass cluster with the CAPD provider.
```

Comment:
The CAPD provider is the key part here though - I'd like to really highlight what this guide does - and hopefully we can expand it for other providers sooner rather than later.